### ISSUE TITLE
Refactor: replace CLI dependency with a commonly shared library instead

### ISSUE THREAD

**Issue by gr2m** [all-contributors/app#434]
follow up to https://github.com/all-contributors/app/issues/429#issuecomment-1270550950

We can discuss details here. I think we had discussions about this in the past but I'm not sure where. @Berkmann18 do you remember?

**Comment by Berkmann18** [comment]
@gr2m I tried finding it, in the discussions (which was where we discussed that IIRC) or the references on https://github.com/all-contributors/architecture/issues/3 but I can't find it.

**Comment by lwasser** [comment]
oooh @Berkmann18 is this what you were talking about with infrastructure refactoring ? @gr2m we met last week and Max mentioned refactoring / cleaning up the architecture. I see a lot of issues related to the CLI being ahead of the app. So doing this would simplify things (if i understand the setup correctly). i'll flag this one as infra and high priority. it seems like it might fix a lot of issues with syncing the two.

relates to #507

**Comment by gr2m** [comment]
yes, we both should work with the same code base. It would simplify a lot, most importantly reduce maintenance overhead in future

**Comment by lwasser** [comment]
@gr2m , I have another question for you. (Be patient with me as I get to know the infrastructure here) - @JoshuaKGoldberg submitted this pr  https://github.com/all-contributors/app/pull/508/ that updates the CLI for the app. But the CLI is NOT a library as it currently stands. Is that correct? I'm a Python person, so I'll be learning JS and asking questions as I go along here. @JimMadge, including you here too. i think this is an area of focus for us moving forward.

**Comment by gr2m** [comment]
> But the CLI is NOT a library as it currently stands

It is both a CLI and it does export APIs that can be consumed when using `all-contributors-cli` as a dependency. But I'd argue that is bad design. The exported methods are in this file:

https://github.com/all-contributors/cli/blob/master/src/api.js

We even log a warning when using the CLI this way.

The correct design would be to have a shared `all-contributors` library, which then is consumed by both the CLI and the app.
